lib/src/realtime_db_pagination.dart

```diff
@@ -203,7 +209,9 @@ class _RealtimeDBPaginationState extends State<RealtimeDBPagination> {
     if (getMore) setState(() => _isFetching = true);
 
     final docsLimit = _data.length + (getMore ? widget.limit : 0);
-    var docsQuery = widget.query.limitToFirst(docsLimit);
+    var docsQuery = widget.descending
+        ? widget.query.limitToLast(docsLimit)
+        : widget.query.limitToFirst(docsLimit);
     if (_data.isNotEmpty) {
       docsQuery = docsQuery.startAt(
         Map<String, dynamic>.from(
```

Comment:
Do you think we also need to update the logic here?

Like `startAt` should be `endBefore` and `_data.first.value` should be `_data.last.value`?

Can you map out how the queries will work to decide if the change is needed.

Reply:
During testing, withuout changing `_data.first.value` I tested by replacing `startAt ` with `endBefore` and it did not work.

Reply:
Can you try all the patterns and tell me which one works correctly:

|           | **\_data.first.value** | **\_data.last.value** |
| --------- | ------------------ | ----------------- |
| **startAt**   |                    |                   |
| **endBefore** | ❌                 |                   |

Reply:
|  | _data.first.value | _data.last.value |
| ------------- | ------------- | ------------- |
| startAt  | ✅ | ✅  |
| endBefore  | ❌  | ❌  |

Reply:
Looks like we have to use `_data.last.value` and `endAt` for descending type.